packages/noya-state/src/selectors.ts

```diff
@@ -134,6 +134,20 @@ export const getSelectedStyles = (state: ApplicationState): Sketch.Style[] => {
     : getSelectedLayerStyles(state).map((style) => style.value);
 };
 
+export const getSelectedTextStyles = (
+  state: ApplicationState,
+): Sketch.Style[] => {
+  return getSelectedLayers(state)
+    .filter((layer) => layer._class === 'text')
+    .flatMap((layer) => (layer.style ? [layer.style] : []));
+};
+
+export const getSelectedText = (state: ApplicationState): Sketch.Text[] => {
+  return getSelectedLayers(state)
+    .filter((layer) => layer._class === 'text')
```

Comment:
We can use a type guard here to avoid the map:

```ts
return getSelectedLayers(state)
  .filter((layer): layer is Sketch.Text => layer._class === 'text')
```

Reply:
And rename to `getSelectedTextLayers`